eth/db/storage.py

```diff
@@ -222,7 +213,16 @@ def clear_changed_root(self) -> None:
         self._starting_root_hash = None
 
     def commit_to(self, db: BaseDB) -> None:
+        """
+        Trying to commit changes when nothing has been written will raise a
+        ValidationError
+        """
         self.logger.debug2('persist storage root to data store')
+        if self._trie_nodes_batch is None:
```

Comment:
Sorry for critiquing even the things you add in response to my critiques :sweat_smile: 

But this check isn't quite right. It's possible for `_trie_nodes_batch` to be not none at the same time as there are no pending changes, `commit_to()` doesn't set `_trie_nodes_batch` to `None` so two calls to `commit_to()` right after each other wouldn't trigger this error.

Reply:
Haha, no apologizing for pointing out code that looks bad to you! If I disagree that it's worth taking care of right away, we can always punt to an issue.

But yeah, I agree :). So the next change is:
- set `_trie_nodes_batch = None` in `clear_changed_root()`
- call `clear_changed_root()` at the end of `commit_to()`, instead of having the caller to manage that. No reason the caller needs to handle it AFAICT
- move `clear_changed_root()` to `_clear_changed_root()` since it's never called outside the class anymore